### PR TITLE
Added Pagination for get messages endpoint.

### PR DESCRIPTION
**Pull request by man-of-mind**
Modified the db.py file to raise an exception when there is no internet connection, modified the update message endpoint, it failed one of the test cases.

Kindly review bosses @o4codes @zxenonx @AI-fae @mykie88 @pauline-banye @kinglighthill

### CODE REVIEW

**Comment by zxenonx** [comment]
@man-of-mind, thanks for this great work.
Nevertheless, this PR must be split into separate PRs. It will be very hard to properly review this PR, it contains features, refactoring and tests.

Kindly consider this.

**Comment by man-of-mind** [comment]
> @man-of-mind, thanks for this great work.
> Nevertheless, this PR must be split into separate PRs. It will be very hard to properly review this PR, it contains features, refactoring and tests.
> 
> Kindly consider this. 

Okay boss @zxenonx

**Comment by o4codes** [comment]
Well I agree with @zxenonx

File: backend/endpoints/messages.py [hunk 1]
```diff
@@ -232,7 +233,7 @@ async def get_messages(org_id, room_id):
             status_code=status.HTTP_424_FAILED_DEPENDENCY,
             detail="Zc Core failed",
         )
-    return JSONResponse(
-        content=ResponseModel.success(data=response, message="Messages retrieved"),
-        status_code=status.HTTP_200_OK,
-    )
+    return paginate(response)
```
Comment (paulinebanye) [inline backend/endpoints/messages.py:236]: Well done @man-of-mind it seems there isnt a limit for the paginated results specified in this PR. What's the FastAPI default?

**Review by paulinebanye — APPROVED** [review]

File: backend/endpoints/messages.py (hunk 1 above)
Comment (man-of-mind) [reply backend/endpoints/messages.py:236]: The default is 50, I can add custom limit if it is required.

Comment (paulinebanye) [reply backend/endpoints/messages.py:236]: Oh the default is fine 👍🏻

File: backend/endpoints/messages.py [hunk 2]
```diff
@@ -209,7 +210,7 @@ async def update_message(
 
 @router.get(
     "/org/{org_id}/rooms/{room_id}/messages",
-    response_model=ResponseModel,
+    response_model=Page[Message],
```
Comment (mikenrowland) [inline backend/endpoints/messages.py:213]: boss @man-of-mind abeg how is this used?

**Review by mikenrowland — APPROVED** [review]
LGTM

File: backend/endpoints/messages.py (hunk 2 above)
Comment (man-of-mind) [reply backend/endpoints/messages.py:213]: It's to give custom response model based on the pagination, it's pydantic model that represent the result or response.
@mykie88

File: backend/utils/db.py [hunk 3]
```diff
@@ -28,6 +30,9 @@ def __init__(self, organization_id: str) -> None:
             self.plugin_id = plugin["id"] if plugin else None
         except requests.exceptions.RequestException as exception:
             print(exception)
+            raise HTTPException from exception(
```
Comment (zxenonx) [inline backend/utils/db.py:33]: Nice

File: backend/endpoints/messages.py [hunk 4]
```diff
@@ -187,7 +188,7 @@ async def update_message(
         )
 
     message["richUiData"] = payload["richUiData"]
-    payload["edited"] = True
+    payload["edited"] = message["edited"] = True
```
Comment (zxenonx) [inline backend/endpoints/messages.py]: Suggestion: 
```
    message.update(payload)
    message['edited'] = True
```

File: backend/endpoints/messages.py [hunk 5]
```diff
@@ -187,7 +188,7 @@ async def update_message(
...
     edited_message = await DB.update(
         MESSAGE_COLLECTION, document_id=message_id, data=payload
```
Comment (zxenonx) [inline backend/endpoints/messages.py]:
```suggestion
        MESSAGE_COLLECTION, document_id=message_id, data=message
```
Because `message` is now updated

**Review by zxenonx — CHANGES_REQUESTED** [review]
Great work @man-of-mind

File: backend/endpoints/messages.py (hunk 4 above)
Comment (man-of-mind) [reply backend/endpoints/messages.py]: Nice one boss

**Review by zxenonx — APPROVED** [review]
Thanks for the changes @man-of-mind

**Pull request merged**